### PR TITLE
Rust activation recipes

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by isuruf** [comment]
@mingwandroid, can you confirm that you want to be a maintainer?

**Comment by mingwandroid** [comment]
For sure I am more than happy to maintain this.

Recently I added rust 1.46.0 to defaults and built the latest of alacritty, ripgrep, nushell and gfx-portability for most OSes. Using these compiler activation scripts mean that our compilers and libraries were used.

The recipes are some of the cleanest and most simple ones I have ever written (which is a testament to cargo and rust more than anything else).

**Comment by mingwandroid** [comment]
.. and the same recipes work for both MSVC and mingw-w64 based rust (though we probably want to think about whether binaries should end up in the same place with each of these toolchains or not, and what to do about that .. a viable option would be sticking with the MSVC rust compiler only which does generate significantly smaller binaries in my experience). For now on defaults, I have provided both.

**Comment by isuruf** [comment]
Thanks @mingwandroid

@conda-forge/staged-recipes, this is ready for review

File: recipes/rust-activation/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2015-2018, conda-forge
```
Comment (xhochy) [inline recipes/rust-activation/LICENSE.txt:1]: Should the date be updated?

**Comment by mingwandroid** [comment]
There are no reviewers. LGTM though.

File: recipes/rust-activation/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+{% set version = "1.46.0" %}
+
+{% if cross_target_platform is undefined %}
```
Comment (xhochy) [inline recipes/rust-activation/meta.yaml:3]: Shouldn't this be always defined?

File: recipes/rust-activation/LICENSE.txt (hunk 1 above)
Comment (isuruf) [reply recipes/rust-activation/LICENSE.txt:1]:
```suggestion
Copyright (c) 2020, conda-forge contributors
```

File: recipes/rust-activation/meta.yaml (hunk 2 above)
Comment (isuruf) [reply recipes/rust-activation/meta.yaml:3]: It should, but for some phases of conda-build is not, and this is a workaround.

**Review by xhochy — APPROVED** [review]

**Comment by jakirkham** [comment]
cc @conda-forge/rust

**Pull request merged**